util/conditions/v1beta2/paused.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/annotations"
+)
+
+type pausedConditionSetter interface {
+	Setter
+	metav1.Object
+}
+
+// SetPausedCondition sets the paused condition on the object and returns if it should be considered as paused.
+func SetPausedCondition[T pausedConditionSetter](cluster *clusterv1.Cluster, ms T) (isPaused bool) {
+	if annotations.IsPaused(cluster, ms) {
+		msgs := []string{}
+		if annotations.HasPaused(ms) {
```

Comment:
@chrischdi I would say let's get consensus on https://github.com/kubernetes-sigs/cluster-api/pull/11276
how the pausee condition should be exactly calculated. Then we can sync it back here once the Machine controller PR is merged